### ISSUE TITLE
Add mockito, bytebuddy, objenesis version to Platform (maybe git commit to)

### ISSUE THREAD

**Issue by bric3** [mockito/mockito#687]
Currently `Platform` lists various versions and is only used in the mockmaker, however this tool be improved to include Mockito version, the commit ID, bytebyddy, objenesis.

Eventually it could also detect version mismatch on the classpath, of those dependencies.

This ticket is a bit vague as the idea has to mature a bit. Ideas welcome ;)

**Comment by szczepiq** [comment]
Interesting. Seems useful :) Let's see what real use cases we encounter to back it up. Platform is an internal API currently.

**Comment by bric3** [comment]
Actually it is for us to diagnose classpath issues, or bad versions. The idea is still vague. I don't know yet if `Platform` can be made non internal at that moment.

One thing to consider though is adding it to every `MockitoException`. Could be via the `Reporter` or not...

**Comment by szczepiq** [comment]
Cool. Interesting idea.

**Comment by ChristianSchwarz** [comment]
> One thing to consider though is adding it to every MockitoException. Could be via the Reporter or not...

The idea is great to aid trouble shooting :+1: !

I think there should be a new API like `printDebugInfos()` or `printPlatformInfos()`. Adding such information on every MockitoException generates maybe too much noise.

**Comment by szczepiq** [comment]
>Adding such information on every MockitoException generates maybe too much noise.

Right, for "expected" exceptions :) like typical verification errors we would not print this.
